src/cli/options/aspa.rs

```diff
@@ -0,0 +1,192 @@
+        self, client: &KrillClient
+    ) -> Result<api::Success, httpclient::Error> {
+        client.aspas_update(
+            &self.ca.ca,
+            api::AspaDefinitionUpdates::new(vec![self.aspa.0], vec![])
+        ).await
+    }
+}
+
+
+//------------ Remove ----------------------------------------------------
+
+#[derive(clap::Args)]
+pub struct Remove {
+    #[command(flatten)]
+    ca: ca::Handle,
+
+    /// Customer ASN of the ASPA to remove
+    #[arg(long, value_name = "ASN")]
+    customer: Asn,
+}
+
+impl Remove {
+    async fn run(
+        self, client: &KrillClient
+    ) -> Result<api::Success, httpclient::Error> {
+        client.aspas_update(
+            &self.ca.ca,
+            api::AspaDefinitionUpdates::new(vec![], vec![self.customer])
+        ).await
+    }
+}
+
+
+//------------ Update --------------------------------------------------------
+
+#[derive(clap::Args)]
+pub struct Update {
+    #[command(flatten)]
+    ca: ca::Handle,
+
+    /// Customer ASN of an existing ASPA
+    #[arg(long, value_name = "ASN")]
+    customer: Asn,
+
+    /// Provider ASN to add
+    #[arg(long, value_name = "ASN")]
+    add: Vec<Asn>,
+
+    /// Provider ASN to remove.
+    #[arg(long, value_name = "ASN")]
+    remove: Vec<Asn>,
+}
+
+impl Update {
+    pub async fn run(
+        self, client: &KrillClient
+    ) -> Result<api::Success, httpclient::Error> {
+        client.aspas_update_single(
+            &self.ca.ca, self.customer,
+            api::AspaProvidersUpdate::new(self.add, self.remove)
+        ).await
+    }
+}
+
+
+//============ Argument Parser Types =========================================
+
+//------------ CleanAspaDefinition -------------------------------------------
+
+#[derive(Clone, Debug)]
+pub struct CleanAspaDefinition(api::AspaDefinition);
+
+impl FromStr for CleanAspaDefinition {
+    type Err = CleanAspaDefinitionError;
+
+    fn from_str(src: &str) -> Result<Self, Self::Err> {
+        let aspa = api::AspaDefinition::from_str(src).map_err(
+            CleanAspaDefinitionError::Format
+        )?;
+        if aspa.customer_used_as_provider() {
+            Err(CleanAspaDefinitionError::CustomerAsProvider)
+        }
+        else if aspa.providers().is_empty() {
+            Err(CleanAspaDefinitionError::EmptyProviders)
+        }
+        else {
+            Ok(Self(aspa))
+        }
+    }
+}
+
+
+//============ ErrorTypes ====================================================
+
+//------------ CleanAspaDefinitionError --------------------------------------
+
+#[derive(Debug)]
+pub enum CleanAspaDefinitionError {
+    Format(api::AspaDefinitionFormatError),
+    CustomerAsProvider,
+    EmptyProviders,
+}
+
+impl fmt::Display for CleanAspaDefinitionError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Format(err) => err.fmt(f),
+            Self::CustomerAsProvider => {
+                f.write_str("Customer AS may not be used as provider.")
```

Comment:
Higher up in this file the references all seem to refer to "ASN" but here the term "AS" is used. Is this correct?